src/rpc/request.cpp

```diff
@@ -82,33 +81,40 @@ static fs::path GetAuthCookieFile(bool temp=false)
 
 static bool g_generated_cookie = false;
 
-bool GenerateAuthCookie(std::string *cookie_out)
+bool GenerateAuthCookie(std::string* cookie_out, std::optional<fs::perms> cookie_perms)
 {
     const size_t COOKIE_SIZE = 32;
     unsigned char rand_pwd[COOKIE_SIZE];
     GetRandBytes(rand_pwd);
     std::string cookie = COOKIEAUTH_USER + ":" + HexStr(rand_pwd);
 
-    /** the umask determines what permissions are used to create this file -
-     * these are set to 0077 in common/system.cpp.
```

Comment:
In 7ad5349a2ec04399c99c8cbf71c8d3b4627132f8 "init: add option for rpccookie permissions"

This comment is removed, but the umask is still what sets the permissions in the default case.

Since we can set the permissions explicitly in this function, I think it would make sense to always set them with `owner` as default rather than relying on the umask that happens somewhere else.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/28167#discussion_r1655290944

> This comment is removed, but the umask is still what sets the permissions in the default case.

Good catch, it would be good to add back the comment.

> Since we can set the permissions explicitly in this function, I think it would make sense to always set them with `owner` as default rather than relying on the umask that happens somewhere else.

The PR was actually doing this originally, but I suggested doing the opposite in https://github.com/bitcoin/bitcoin/pull/28167#discussion_r1628110701. It makes sense to me that when `-rpccookieperms` option is not specified, bitcoind would not set permissions on this file, just like it is not setting permissions on other files.

It seems fragile to rely on the [fs::permissions](https://en.cppreference.com/w/cpp/filesystem/permissions) call doing the right thing on all filesystems, when its behavior is not specified anywhere and we have never relied on it before. The documentation points to some odd ways it can be behave like "changing some bits may automatically change others" and we don't know what unusual things it may do on fuse filesystems or filesystems that use ACLs.

Reply:
I've un-removed the comment in 9eff5601059, as I agree with @ryanofsky that not changing the existant behaviour is the "safer" default, and new behaviour can be introduced with the new option.